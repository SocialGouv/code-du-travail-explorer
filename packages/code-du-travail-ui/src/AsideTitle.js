import React from "react";
import PropTypes from "prop-types";

const AsideTitle = ({ style, children }) => {
  return (
    <h3 className="aside-title" style={style}>
      {children}
    </h3>
  );
};

AsideTitle.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.string
  ]).isRequired,
  style: PropTypes.object
};

export default AsideTitle;
