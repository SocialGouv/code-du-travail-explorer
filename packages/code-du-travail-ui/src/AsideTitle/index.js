import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { colors, fonts, spacing } from "../theme";

const AsideTitle = ({ children }) => {
  return <H3>{children}</H3>;
};

AsideTitle.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.string
  ]).isRequired,
  style: PropTypes.object
};

export default AsideTitle;

const H3 = styled.h3`
  padding: ${spacing.interComponent} 0;
  font-size: ${fonts.sizeH5};
  font-weight: 600;
  color: ${colors.black};
`;
