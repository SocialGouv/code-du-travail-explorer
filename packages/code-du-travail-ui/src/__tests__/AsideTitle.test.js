import React from "react";
import { render } from "react-testing-library";
import AsideTitle from "../AsideTitle";

describe("<AsideTitle />", () => {
  test("should render", () => {
    const { container } = render(<AsideTitle>One Aside title</AsideTitle>);
    expect(container).toMatchSnapshot();
  });
});
