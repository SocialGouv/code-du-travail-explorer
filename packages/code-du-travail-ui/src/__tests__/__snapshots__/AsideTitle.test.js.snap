// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AsideTitle /> should render 1`] = `
<div>
  <h3
    class="aside-title"
  >
    One Aside title
  </h3>
</div>
`;
